sycl/include/sycl/ext/oneapi/matrix/matrix-unified.hpp

```diff
@@ -0,0 +1,221 @@
+//===------- matrix-unified.hpp - SYCL matrix extension ----*- C++ -*------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// ===--------------------------------------------------------------------=== //
+
+#pragma once
+#include <sycl/ext/oneapi/matrix/matrix-tensorcores.hpp>
+
+namespace sycl {
+__SYCL_INLINE_VER_NAMESPACE(_V1) {
+namespace ext {
+namespace oneapi {
+namespace experimental {
+namespace matrix {
+
+template <typename Group, typename T, use Use, size_t Rows, size_t Cols,
+          layout Layout>
+struct joint_matrix {
+
+#if defined(__SYCL_DEVICE_ONLY__) && defined(__SPIR__)
```

Comment:
sorry, i remember previously it is:
```
#if defined(__SYCL_DEVICE_ONLY__)
#if defined(__NVPTX__)
  sycl::ext::oneapi::detail::joint_matrix_cuda<T, Use, Rows, Cols, Layout>
      cuda_impl;
#else
  __spv::__spirv_JointMatrixINTEL<
      T, Rows, Cols, spv_matrix_layout_traits<Layout>::value,
      spv_scope_traits<Group>::value, spv_matrix_use_traits<Use>::value> *spvm;
#endif // defined(__SYCL_DEVICE_ONLY__)
#endif
```
in intel side, we can't let host compilation use sycl::ext::oneapi::detail::joint_matrix_cuda.
could we change back preivous code?